compiler/next/include/chpl/resolution/resolution-types.h

```diff
@@ -20,45 +20,71 @@
 #ifndef CHPL_RESOLUTION_RESOLUTION_TYPES_H
 #define CHPL_RESOLUTION_RESOLUTION_TYPES_H
 
-#include "chpl/uast/TypeDecl.h"
+#include "chpl/types/Type.h"
+#include "chpl/uast/ASTNode.h"
 
 namespace chpl {
 namespace resolution {
 
 
-class Type {
- public:
-  enum TypeTag {
-    FROM_DECL,
-    INT8,
-    INT16,
-    INT32,
-    INT64,
-    UINT8,
-    UINT16,
-    UINT32,
-    UINT64,
-    BOOL,
-    BOOL8,
-    BOOL16,
-    BOOL32,
-    BOOL64,
-    REAL32,
-    REAL64,
-    IMAG32,
-    IMAG64,
-    COMPLEX64,
-    COMPLEX128,
-  };
- private:
-  TypeTag tag_;
-  const uast::TypeDecl* typeDecl_;
- public:
-  TypeTag tag() { return tag_; } 
-  const uast::TypeDecl* typeDecl() { return typeDecl_; }
-  UniqueString name(Context* context);
+struct ResolutionResult {
+  // the expr that is resolved
+  const uast::Expression* expr = nullptr;
+  // For simple cases, which named decl does it refer to?
+  const uast::NamedDecl* decl = nullptr;
+  // What is its type?
+  const types::Type* type = nullptr;
+  // For a function call, it might refer to several Functions
+  // and we might not know which return intent to choose yet.
+  std::vector<const uast::Function*> otherFns;
```

Comment:
Could we name this `candidates`?

Reply:
Sure, will happen in a future PR though.